drivers/ethernet/eth_sam_gmac.c

```diff
@@ -1306,6 +1332,12 @@ static int eth_tx(struct device *dev, struct net_pkt *pkt)
 		__ASSERT(tx_desc_list->head != tx_desc_list->tail,
 			 "tx_desc_list overflow");
 
+		/* Account for a sent frag */
+		ring_buf_put(&queue->tx_frag_list, POINTER_TO_UINT(frag));
```

Comment:
if the buffer size is small (let's say 128 bytes), each net_pkt will easily have many "frag".  Is there a risk of the ring buf being full then?

Reply:
Yes, the ring buffer contains as many elements as the number of descriptors. Actually we could just use a simple array accessed with the descriptor number instead of the ring_buf if there was no need to unref all the buffers/packets in case of error.